### PR TITLE
[MPS]: Add fix for squeezed input axes handling in BCE loss

### LINKED ISSUE #79527: torch.nn.BCELoss throws 'invalid axes' error when using "mean" or "sum" reduction on torch.device("mps")
### 🐛 Describe the bug

Computing BCELoss with tensors on torch.device("mps") with reduction gives an 'invalid axes' error:

```
loc("reductionMeanTensor"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/b6051351-c030-11ec-96e9-3e7866fcf3a1/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":281:0)): error: invalid axes: 2, 3
LLVM ERROR: Failed to infer result type(s).
```

The code to reproduce

```Python

import torch

device = torch.device("mps")
a = torch.rand(1, 1, 32, 32)
b = torch.rand(1, 1, 32, 32)
a_mps = a.to(device)
b_mps = b.to(device)

lossfn_no_reduction = torch.nn.BCELoss(reduction="none")
lossfn_w_reduction = torch.nn.BCELoss(reduction="mean")

# OK
lossfn_no_reduction(a_mps, b_mps)

# OK
lossfn_w_reduction(a, b)

# error
lossfn_w_reduction(a_mps, b_mps)
```

However, computing the loss without reduction and then calling `.mean()` or `.sum()` method on the non-reduced tensor works fine.

### Versions

```
Collecting environment information...
PyTorch version: 1.13.0.dev20220614
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.23.2
Libc version: N/A

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:14)  [Clang 12.0.1 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.4
[pip3] torch==1.13.0.dev20220614
[pip3] torchvision==0.14.0.dev20220614
[conda] numpy                     1.22.4           py38he1fcd3f_0    conda-forge
[conda] pytorch                   1.13.0.dev20220614         py3.8_0    pytorch-nightly
[conda] torchvision               0.14.0.dev20220614        py38_cpu    pytorch-nightly
```

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @kulinseth

### PR DESCRIPTION
**Pull request by kulinseth**
Fixes #79527

### CODE REVIEW

**Review by razarmehr — APPROVED** [review]
Lintrunner errors should be fixed.

**Review by albanD — APPROVED** [review]
SGTM
Lint needs fixing though.
